android/src/main/java/com/attentivereactnativesdk/AttentiveReactNativeSdkModule.java

```diff
@@ -0,0 +1,194 @@
+    }
+    if (identifiers.hasKey("klaviyoId")) {
+      idsBuilder.withKlaviyoId(identifiers.getString("klaviyoId"));
+    }
+    if (identifiers.hasKey("shopifyId")) {
+      idsBuilder.withShopifyId(identifiers.getString("shopifyId"));
+    }
+    if (identifiers.hasKey("clientUserId")) {
+      idsBuilder.withClientUserId(identifiers.getString("clientUserId"));
+    }
+    if (identifiers.hasKey("customIdentifiers")) {
+      Map<String, String> customIds = new HashMap<>();
+      Map<String, Object> rawCustomIds = identifiers.getMap("customIdentifiers").toHashMap();
+      for (Map.Entry<String, Object> entry : rawCustomIds.entrySet()) {
+        if (entry.getValue() instanceof String) {
+          customIds.put(entry.getKey(), (String) entry.getValue());
+        }
+      }
+      idsBuilder.withCustomIdentifiers(customIds);
+    }
+
+    attentiveConfig.identify(idsBuilder.build());
+  }
+
+  @ReactMethod
+  public void recordProductViewEvent(ReadableMap productViewAttrs) {
+    Log.i(TAG, "Sending product viewed event");
+
+    List<Item> items = buildItems(productViewAttrs.getArray("items"));
+    ProductViewEvent productViewEvent = new ProductViewEvent.Builder(items).build();
+
+    AttentiveEventTracker.getInstance().recordEvent(productViewEvent);
+  }
+
+  @ReactMethod
+  public void recordPurchaseEvent(ReadableMap purchaseAttrs) {
+    Log.i(TAG, "Sending purchase event");
+    Order order = new Order.Builder(purchaseAttrs.getMap("order").getString("orderId")).build();
+
+    List<Item> items = buildItems(purchaseAttrs.getArray("items"));
+    PurchaseEvent purchaseEvent = new PurchaseEvent.Builder(items, order).build();
+
+    AttentiveEventTracker.getInstance().recordEvent(purchaseEvent);
+  }
+
+  @ReactMethod
+  public void recordAddToCartEvent(ReadableMap addToCartAttrs) {
+    Log.i(TAG, "Sending add to cart event");
+
+    List<Item> items = buildItems(addToCartAttrs.getArray("items"));
+    AddToCartEvent addToCartEvent = new AddToCartEvent.Builder(items).build();
+
+    AttentiveEventTracker.getInstance().recordEvent(addToCartEvent);
+  }
+
+  @ReactMethod
+  public void recordCustomEvent(ReadableMap customEventAttrs) {
+    Log.i(TAG, "Sending custom event");
+    ReadableMap propertiesRawMap = customEventAttrs.getMap("properties");
+    if (propertiesRawMap == null) {
+      throw new IllegalArgumentException("The CustomEvent 'properties' field cannot be null.");
+    }
+    Map<String, String> properties = convertToStringMap(propertiesRawMap.toHashMap());
+    CustomEvent customEvent = new CustomEvent.Builder(customEventAttrs.getString("type"), properties).build();
+
+    AttentiveEventTracker.getInstance().recordEvent(customEvent);
+  }
+
+  private Map<String, String> convertToStringMap(Map<String, Object> inputMap) {
+    Map<String, String> outputMap = new HashMap<>();
+    for (Map.Entry<String, Object> entry : inputMap.entrySet()) {
+      outputMap.put(entry.getKey(), (String) entry.getValue());
+    }
```

Comment:
will this handle null values or should we filter those out

Reply:
I added null checking that throws an error. Also filters non-strings out.